collection_size.go

```diff
@@ -5,30 +5,86 @@ package monitoring
 
 import (
 	"fmt"
+	"sync"
 
+	"github.com/juju/errors"
 	"github.com/juju/loggo"
 	"github.com/prometheus/client_golang/prometheus"
 	"gopkg.in/mgo.v2"
 	"gopkg.in/mgo.v2/bson"
 )
 
 var (
-	log = loggo.GetLogger("collectors")
+	log = loggo.GetLogger("monitoring.collectors")
 )
 
+type stats struct {
+	Count int32 `bson:"count"`
+	Size  int32 `bson:"size"`
+}
+
+type statsGetter struct {
+	session *mgo.Session
+	closed  bool
+	lock    sync.Mutex
```

Comment:
It's customary to declare the mutex before the members that are protected by it in a struct.
https://talks.golang.org/2014/readability.slide#21